File: python/src/core/models/entities/key.py
from pydantic import Field
from typing import Optional
from datetime import datetime
from models.entities.abstract.named_entity import NamedEntity


class Key(NamedEntity):
    # ... means required in Pydantic
    host_id: str = Field(..., alias="host_id")
    is_active: bool = Field(default=True, alias="is_active")
    created_date: Optional[datetime] = Field(
        default=None, alias="created_date")

    class Config:
        allow_population_by_field_name = True

File: core/sdk/dotnet/Models/Entities/Key.cs
using Agience.Core.Models.Entities.Abstract;
using System.Text.Json.Serialization;

namespace Agience.Core.Models.Entities
{
    public class Key : NamedEntity
    {
        [JsonPropertyName("host_id")]
        public string HostId { get; set; } = null!;

        [JsonPropertyName("is_active")]
        public bool IsActive { get; set; } = true;

        [JsonPropertyName("created_date")]
        public DateTime? CreatedDate { get; set; } = null;
    }
}

